### PR TITLE
fix: [BootGuardLibrary] TPM intialization for ME less platform

### PR DESCRIPTION
**Pull request by aborwank**
added support for server platfrom which do not have ME to skip the ME FW status check by comparing HeciBaseAddress.

### CODE REVIEW

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c [hunk 1]
```diff
@@ -78,18 +78,23 @@ GetBootGuardInfo (
...
     DEBUG ((DEBUG_INFO, "Boot Guard Boot Status = %llx\n", BootGuardBootStatus));
 
+    DEBUG ((DEBUG_INFO, "HeciBaseAddress = %llx\n", HeciBaseAddress));
     ///
     /// Read ME FWS Registers
     ///
-    MeFwSts4 = MmioRead32 (HeciBaseAddress + R_ME_HFS_4);
-    DEBUG ((DEBUG_INFO, "ME FW STS 4 = %x\n", MeFwSts4));
+    if (HeciBaseAddress != 0) {
+      MeFwSts4 = MmioRead32 (HeciBaseAddress + R_ME_HFS_4);
```
Comment (lsubash) [inline Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c:86]: Would you add comments when HeciBaseAddress is expected to 0x0?

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c [hunk 2]
```diff
@@ -78,18 +78,23 @@ GetBootGuardInfo (
...
+      MeFwSts4 = 0x0;
 
     ///
     /// Check Bit 12 in ME FWSTS4 to check if TPM_DISCONNECT_ALL bit is set
     /// or ENF Shutdown path is taken by ME FW.
     /// Also Identify any failures in ACM
     ///
-    if (((MeFwSts4 & (B_TPM_DISCONNECT | B_BOOT_GUARD_ENF_MASK)) != 0) ||
+    if ((((MeFwSts4 & (B_TPM_DISCONNECT | B_BOOT_GUARD_ENF_MASK))
+        && HeciBaseAddress) != 0) ||
```
Comment (lsubash) [inline Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c:97]: HeciBaseAddress need not be checked as MeFwSts4 will be zero.

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c [hunk 3]
```diff
@@ -105,7 +110,7 @@ GetBootGuardInfo (
       BootGuardInfo->TpmStartupFailureOnS3 = TRUE;
       DEBUG ((DEBUG_INFO, "ACM Tpm2Startup (State) failure detected during S3 flow.\n"));
     }
-    if ((MeFwSts4 & BIT10) != 0) {
+    if (((MeFwSts4 & BIT10) && HeciBaseAddress) != 0) {
```
Comment (lsubash) [inline Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c:113]: Same here.

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c (hunk 1 above)
Comment (aborwank) [reply Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c:88]: done

**Review by aborwank — COMMENTED** [review]
Please merge

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c (hunk 2 above)
Comment (aborwank) [reply Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c]: done

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c (hunk 3 above)
Comment (aborwank) [reply Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardLibrary.c]: done

**Review by lsubash — APPROVED** [review]

**Pull request merged**